A small turbulent velocity script

fjebaker opened the pull request:
Yields the following figures:

![image](https://github.com/user-attachments/assets/b81475f1-7f53-4397-b634-247c85863532)

![image](https://github.com/user-attachments/assets/4cb255b1-8bd3-4ffa-99f3-7805ce51c1cb)

![image](https://github.com/user-attachments/assets/f41857b6-5e0e-476c-86b8-69d8b4a00409)

![image](https://github.com/user-attachments/assets/a472737e-a639-4851-b509-f7e19ef3e2c5)

fjebaker committed:
feat: added turbulent velocity visualiser

Added a small script that lets you implement a turbulent velocity
function and compare it to the Keplerian equivalent using a variety of
plots.

phajy commented:
Thanks, @fjebaker - this looks great! @katebolam and @jj-auer you should be able to see the section of the code where you can put in a prescription for the turbulence.

This reminded me that perhaps §2 of [Accretion disc turbulence and the X-ray power spectra of black hole high states](https://ui.adsabs.harvard.edu/abs/1995MNRAS.274...37N/abstract) (1995) by Nowak & Wagoner might be worth a read where they describe the "basics of disc turbulence".

However, I would recommend starting with a pretty simple description just to get things working and to make some nice plots for your presentation. You might also want to investigate how to use [Gradus.jl](https://github.com/astro-group-bristol/Gradus.jl) with this new velocity field (we can talk about this at the next project meeting).

One possible Julia package for generating noise could be [CoherentNoise.jl](https://github.com/lazarusA/CoherentNoise.jl). Alternatively, you could write your own, which might become necessary to really understand how the noise is being generated and to ensure it has the desired properties. Any other suggestions, @fjebaker?

Happy for this to be merged once Kate and Joschka have had a look.

fjebaker committed:
Updated the size of the turbulent noise

fjebaker committed:
Merge pull request #2 from phajy/fergus/demo

Scaling turbulent velocities

fjebaker committed:
chore: refactor velocity structures

fjebaker committed:
feat: add ray-traced turbulent example

I had to remove the condition on the velocity constraint, as the ad-hoc
method of generating noise makes the velocity components invalid.

fjebaker committed:
chore: add Project.toml

fjebaker committed:
feat: add lineprofile turbulent example
